pkg/api/controller.go

```diff
@@ -3006,6 +3025,69 @@ func (c *Controller) GetCurrentUser(w http.ResponseWriter, r *http.Request) {
 	writeResponse(w, http.StatusOK, response)
 }
 
+func (c *Controller) sendResetPasswordEmail(email string, token string) error {
+	if c.Emailer.Limiter.Allow() {
+		return c.Emailer.SendEmail([]string{email}, token, token, nil)
+	}
+	err := ErrRateLimitExceeded
+	c.Logger.WithError(err).WithField("email", email).Debug("Email sending rate limit was exceeded")
+	return err
+}
```

Comment:
The usual style for early-return is to  return quickly on exceptional conditions.  In this case it would be:
```suggestion
	if !c.Emailer.Limiter.Allow() {
		err := ErrRateLimitExceeded
		c.Logger.WithError(err).WithField("email", email).Warn("Email sending rate limit exceeded")
	}
	return c.Emailer.SendEmail([]string{email}, token, token, nil)
}
```

Also we need at least to **warn** here: this is a condition that interests operators.

Reply:
fixed